### DISCUSSION TITLE
zerossl 的 ecc 证书在 MT7621A 平台 CPU会100%

### DISCUSSION THREAD

**Discussion by kousyougi** [XTLS/Xray-core#562] [Q&A]
发现一个奇怪的现象，在X86平台完全没问题。

是用acme.sh自动申请的证书。
伺服器端换上 zerossl 的 ec-256 证书 会让我客户端的MT7621A的 CPU满载，网页几乎开不了。
其他 zerossl 的 RSA 证书、Let's Encrypt 的 ec-256证书、RSA证书都没有这个现象，开网页时CPU负载大概是20%~50%。

请问有其他人伺服器用zerossl 的 ec-256证书 ，客户端是MT7621A之类比较弱的CPU吗？

**Comment by kousyougi** [comment]
上面那个ECC证书让CPU满载的问题，把伺服器端的fullchain证书换成站点证书，客户端指定使用中间证书(ZeroSSL ECC Domain Secure Site CA)之后就正常了。

有可能是zerossl的证书中间证书有两层，也有可能是zerossl 的 Certification Paths 有两个造成的。
我完全不懂ssl证书运作的原理，只在找他跟lets encrypt的差异。

![zerossl](https://user-images.githubusercontent.com/43228565/120322703-48531280-c317-11eb-80fc-f57a0b8b3811.jpg)

然后我又做了其他测试，
发现在MT7621A下，跑xtls，ECC速度跟RSA证书两者的速度跟CPU的差距不小。

CPU：
![CPU](https://user-images.githubusercontent.com/43228565/120321694-315ff080-c316-11eb-91f6-f5f588223db0.jpg)
ECC测速
![ECC](https://user-images.githubusercontent.com/43228565/120324028-b77d3680-c318-11eb-8dd9-0749480e0929.jpg)
RSA测速
![RSA](https://user-images.githubusercontent.com/43228565/120321722-3a50c200-c316-11eb-950c-6a393f8c0453.jpg)

**Comment by junmin** [comment]
谢谢提醒，我把服务器的证书换成了lesencrypt的fullchain证书就没有问题了，不知道为什么zerossl会有这个问题。

https://github.com/XTLS/Xray-core/issues/491

**Comment by 8884361** [comment]
留一下记录。
